### PR TITLE
[NTUSER] Redraw issues on window without draw background

### PR DESCRIPTION
**Pull request by turican0**
## Purpose

Some apps don't render the background on startup and don't set the rendering area correctly. The application relies on the OS to initialize the window on startup. This initialization is missing in ReactOS.

JIRA issue: [CORE-18800](https://jira.reactos.org/browse/CORE-18800)

## Proposed changes

I added WM_CREATE to the background rendering section of defwnd.c.

## TODO

_Use a TODO when your pull request is Work in Progress._

- [ ] 
- [ ]

### CODE REVIEW

**Comment by turican0** [comment]
image

**Comment by turican0** [comment]
Neither of the associated problems are affected, they are similar problems and the size of the rendering area initialization, but not directly related. 
[CORE-17678](https://jira.reactos.org/browse/CORE-17678)
[CORE-18800](https://jira.reactos.org/browse/CORE-18800)

**Comment by JoachimHenze** [comment]
Looks promising, but is `return 1` the correct retval for WM_CREATE?
I do suggest performing a testbot run before doing the merge.

**Comment by oleg-dubinskiy** [comment]
> Looks promising, but is `return 1` the correct retval for WM_CREATE? I do suggest performing a testbot run before doing the merge.

https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-create
> If an application processes this message, it should return zero to continue creation of the window. If the application returns –1, the window is destroyed and the [CreateWindowEx](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-createwindowexa) or [CreateWindow](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-createwindowa) function returns a NULL handle.

Nothing is said there about `1`. So yes, seems it isn't.

**Comment by turican0** [comment]
I'll check the API tests, maybe it's better to change it to this (and I'm still wondering if the condition with CS_PARENTDC is important):
```
      case WM_CREATE:
      {
         RECT Rect;
         HBRUSH hBrush = Wnd->pcls->hbrBackground;
         if (!hBrush) return 0;
         if (hBrush <= (HBRUSH)COLOR_MENUBAR)
         {
            hBrush = IntGetSysColorBrush(HandleToUlong(hBrush));
         }
         if (Wnd->pcls->style & CS_PARENTDC)
         {
            /* can't use GetClipBox with a parent DC or we fill the whole parent */
            IntGetClientRect(Wnd, &Rect);
            GreDPtoLP((HDC)wParam, (LPPOINT)&Rect, 2);
         }
         else
         {
            GdiGetClipBox((HDC)wParam, &Rect);
         }
         FillRect((HDC)wParam, &Rect, hBrush);
         return (0);
      }
	  
      case WM_ERASEBKGND:
      case WM_ICONERASEBKGND:
```

**Comment by turican0** [comment]
**I've tried variations:**

-without patch
-s return 1 (WM_CREATE for WM_ERASEBKGND and WM_ICONERASEBKGND)
-s return 0 (WM_CREATE alone)

**and for all variants the 3 key apitests showed the same results:**
CreateWindowEx 58/472
ShowWindow 0/235
NtUserCreateWindowEx 2/64

But I'd rather modify it to a separate variant with return 0 anyway;

**Comment by turican0** [comment]
Anyway, this only solved part of the problems with HexEditor. But that doesn't mean that this fix is unnecessary, in my apitests(I'll add them with a patch soon) windows are also rendered without background in certain cases.

File: win32ss/user/ntuser/defwnd.c [hunk 1]
```diff
@@ -1068,6 +1068,29 @@ IntDefWindowProc(
...
+            /* can't use GetClipBox with a parent DC or we fill the whole parent */
+            IntGetClientRect(Wnd, &Rect);
+            GreDPtoLP((HDC)wParam, (LPPOINT)&Rect, 2);
+         }
+         else
+         {
+            GdiGetClipBox((HDC)wParam, &Rect);
+         }
+         FillRect((HDC)wParam, &Rect, hBrush);
+         return (0);
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/defwnd.c:1091]: There is no need to duplicate the whole case code: simply conditionally return the correct value(s).

Comment (oleg-dubinskiy) [reply win32ss/user/ntuser/defwnd.c:1091]: Yeah, that's what I just wanted to point out too.

Comment (turican0) [reply win32ss/user/ntuser/defwnd.c]: Sure, I merged the code into one block, it'll make more sense and clarity in the end.

**Pull request closed without merging**